### ISSUE TITLE
LookBehind Assertion

### ISSUE THREAD

**Issue by privateOmega** [eslint/eslint#11426]
**What rule do you want to change?**
I would appreciate it if someone could make Lookbehind assertions as valid. The look behind assertion was introduced to node.js in 8.10.0 and now the latest release is 11.10.0. 
https://developer.mozilla.org/en-US/docs/Web/JavaScript/Reference/Global_Objects/RegExp#assertions
https://github.com/tc39/proposal-regexp-lookbehind

**Please provide some example code that this change will affect:**

```js
searchQuery.match(/(?<=\bby\s+)(\w+)/gi)
```

**What does the rule currently do for this code?**
error  Parsing error: Invalid regular expression: /(?<=\bby\s+)(\w+)/: Invalid group

**What will the rule do after it's changed?**
No Error.

**Are you willing to submit a pull request to implement this change?**
No. I lack the know-how to implement the change.

**Comment by not-an-aardvark** [comment]
Hi, thanks for creating an issue. Lookbehind assertions were introduced in ES2018, so to enable support for them you should enable parsing support for ES2018 (or ES2019) in your config:

```json
{
  "parserOptions": {
    "ecmaVersion": 2018
  }
}
```

**Comment by aladdin-add** [comment]
Lookbehind was introduced in es2018 (eslint had supported). did you set the right parserOption: `"{ecmaVersion: 2018}"`?

**Comment by privateOmega** [comment]
@not-an-aardvark @aladdin-add Thanks a lot guys. I must have missed it. I am closing the issue.
